connect/api/src/main/java/org/apache/kafka/connect/data/Values.java

```diff
@@ -1039,30 +1041,34 @@ private static SchemaAndValue parseAsNumber(String token) {
                 return new SchemaAndValue(schema, decimal);
             }
         }
+        
+    	public static boolean isBigInteger(BigDecimal bd) {
+    		return bd.signum() == 0 || bd.scale() <= 0 || bd.stripTrailingZeros().scale() <= 0;
```

Comment:
This stripTrailingZeros covers the cases that the ceil/floor comparison was supposed to cover, great find!